hikari/config.py

```diff
@@ -333,7 +333,7 @@ def _(self, _: attr.Attribute[typing.Optional[int]], value: typing.Optional[int]
     _ssl: typing.Union[bool, ssl_.SSLContext] = attr.field(
         default=True,
         converter=_ssl_factory,
-        validator=attr.validators.instance_of(ssl_.SSLContext),  # type: ignore[assignment,arg-type]
+        validator=attr.validators.instance_of((ssl_.SSLContext, bool)),
     )
```

Comment:
This is not correct, as we want to make sure that `_ssl` cannot hold a bool. Unfortunately, due to how attr works, the typehint we provide will be the same as the one used for both the `__init__` and the value hold. This type ignore is the only way we can fix this

Reply:
Are you sure? Because `ssl` has the docstring:
https://github.com/hikari-py/hikari/blob/15624037e97d989c2e22b1bb1e1f8f6afd28f8ac/hikari/config.py#L341-L376

Which mentions it can be assigned either a bool or an SSLContext,

Reply:
Yes, we allow passing both bool and SSLContext. A bool will later be converted into a SSLContext in `_ssl_factory`. The validator just asserts that what was returned was an SSLContext

Reply:
Shouldn't `_ssl` be `ssl_.SSLContext` then? mypy at the very least takes the type of the converter (`_ssl_factory`) for the type in `__init__`, not the type hint: https://github.com/python/mypy/blob/master/mypy/plugins/attrs.py#L113